catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/NitfAttributeImpl.java

```diff
@@ -61,10 +64,26 @@ protected NitfAttributeImpl(final String attributeName, final String shortName,
         this.attributeDescriptors.add(attributeDescriptor);
         if (StringUtils.isNotEmpty(extNitfName)) {
             this.attributeDescriptors.add(ExtNitfUtility.createDuplicateDescriptorAndRename(
-                    extNitfName, attributeDescriptor));
+                    extNitfName,
+                    attributeDescriptor));
         }
     }
 
+    protected NitfAttributeImpl(String extNitfName, String shortName,
+            Function<T, Serializable> accessorFunction, AttributeType attributeType,
+            List<NitfAttribute<TreGroup>> indexedAttributes) {
```

Comment:
So there are some TREs that have fields that can contain TRE groups (which is like a another TRE) within them. "indexedAttributes" represents the fields that can potentially be in the TRE groups.

So for example, the PIAPRD TRE has a field called KEYWORD. That field, could have multiple TRE groups within it. The IndexedPiaprdAttribute is a class that represents the fields in the TRE groups.

I have added a new constructor to allow passing in indexedAttributes, so that fields within TRE groups of a TRE field can be included in the metacard.

I hope that helps explain the significance of indexedAttributes.